cannot inject types named Provider

Using dagger 1.0.0, I want to inject a type named `denominator.Provider`.  Runtime injection works fine, but the compiler accidentally injects `javax.inject.Provider`.  I've confirmed this by renaming `denominator.Provider` to `denominator.Provider1`, which passes.

```
warning: [options] bootstrap class path not set in conjunction with -source 1.6
/Users/adrianc/git/denominator/denominator-core/build/classes/main/denominator/CredentialsConfiguration$CredentialsAsList$$InjectAdapter.java:56: error: constructor CredentialsAsList in class CredentialsAsList cannot be applied to given types;
    CredentialsConfiguration.CredentialsAsList result = new CredentialsConfiguration.CredentialsAsList(provider.get(), sources.get());
                                                        ^
  required: denominator.Provider,Set<Supplier<Credentials>>
  found: javax.inject.Provider,Set<Supplier<Credentials>>
  reason: actual argument javax.inject.Provider cannot be converted to denominator.Provider by method invocation conversion
/Users/adrianc/git/denominator/denominator-core/build/classes/main/denominator/DNSApiManager$$InjectAdapter.java:59: error: constructor DNSApiManager in class DNSApiManager cannot be applied to given types;
    DNSApiManager result = new DNSApiManager(provider.get(), api.get(), closer.get());
                           ^
  required: denominator.Provider,DNSApi,Closeable
  found: javax.inject.Provider,DNSApi,Closeable
  reason: actual argument javax.inject.Provider cannot be converted to denominator.Provider by method invocation conversion
2 errors
1 warning
```

--- a/pom.xml
+++ b/pom.xml
@@ -43,7 +43,7 @@
     <!-- Compilation -->
     <java.version>1.6</java.version>
     <javax.inject.version>1</javax.inject.version>
-    <javawriter.version>1.0.2</javawriter.version>
+    <javawriter.version>1.0.5</javawriter.version>
 
     <!-- Test Dependencies -->
     <junit.version>4.10</junit.version>

